api/tasks_test.go

```diff
@@ -502,6 +516,177 @@ func TestGetTasksHandler(t *testing.T) {
 					So(respTasks.Limit, ShouldEqual, expectedTasks.Limit)
 					So(respTasks.Offset, ShouldEqual, expectedTasks.Offset)
 					So(respTasks.TotalCount, ShouldEqual, expectedTasks.TotalCount)
+
+					Convey("And the etag of the resource should be returned via the ETag header", func() {
+						So(resp.Header().Get(dpresponse.ETagHeader), ShouldNotBeEmpty)
+					})
+				})
+			})
+		})
+	})
+
+	Convey("Given valid etag", t, func() {
+		httpClient := dpHTTP.NewClient()
+		apiInstance := api.Setup(mux.NewRouter(), dataStorerMock, &apiMock.AuthHandlerMock{}, taskNames, cfg, httpClient, &apiMock.IndexerMock{}, &apiMock.ReindexRequestedProducerMock{})
+
+		Convey("When request is made to get tasks", func() {
+			req := httptest.NewRequest(http.MethodGet, fmt.Sprintf("http://localhost:25700/jobs/%s/tasks?offset=%d&limit=%d", validJobID1, validOffset, validLimit), nil)
+			headers.SetIfMatch(req, `"024613bed9e9db8b730c94a98a420c2ef7e39a09"`)
+
+			resp := httptest.NewRecorder()
+			apiInstance.Router.ServeHTTP(resp, req)
+
+			Convey("Then 200 status code should be returned", func() {
+				So(resp.Code, ShouldEqual, http.StatusOK)
+
+				payload, err := io.ReadAll(resp.Body)
+				if err != nil {
+					t.Errorf("failed to read payload with io.ReadAll, error: %v", err)
+				}
+
+				respTasks := models.Tasks{}
+				err = json.Unmarshal(payload, &respTasks)
+				So(err, ShouldBeNil)
+
+				expectedTask1 := expectedTask(cfg, validJobID1, validTaskName1, true, zeroTime, 0)
+				expectedTask2 := expectedTask(cfg, validJobID1, validTaskName2, true, zeroTime, 0)
+				expectedTasks := &models.Tasks{
+					Count:      2,
+					TaskList:   []models.Task{expectedTask1, expectedTask2},
+					Limit:      2,
+					Offset:     0,
+					TotalCount: 2,
+				}
+
+				Convey("And the new task resource should contain expected values", func() {
+					So(respTasks.Count, ShouldEqual, expectedTasks.Count)
+					So(respTasks.TaskList, ShouldResemble, expectedTasks.TaskList)
+					So(respTasks.Limit, ShouldEqual, expectedTasks.Limit)
+					So(respTasks.Offset, ShouldEqual, expectedTasks.Offset)
+					So(respTasks.TotalCount, ShouldEqual, expectedTasks.TotalCount)
+
+					Convey("And the etag of the resource should be returned via the ETag header", func() {
+						So(resp.Header().Get(dpresponse.ETagHeader), ShouldNotBeEmpty)
+					})
+				})
+			})
+		})
+	})
+
+	Convey("Given empty etag", t, func() {
+		httpClient := dpHTTP.NewClient()
+		apiInstance := api.Setup(mux.NewRouter(), dataStorerMock, &apiMock.AuthHandlerMock{}, taskNames, cfg, httpClient, &apiMock.IndexerMock{}, &apiMock.ReindexRequestedProducerMock{})
+
+		Convey("When request is made to get tasks", func() {
+			req := httptest.NewRequest(http.MethodGet, fmt.Sprintf("http://localhost:25700/jobs/%s/tasks?offset=%d&limit=%d", validJobID1, validOffset, validLimit), nil)
+			headers.SetIfMatch(req, "")
```

Comment:
Please handle the SetIfMatch error here too.

Reply:
Done